controller/volume_attachment_controller_test.go

```diff
@@ -1,400 +1,500 @@
+			Generation: 0,
+		},
+	}
+	tc.expectedVol.Spec.NodeID = ""
+	testCases["test case 10: ticket with higher priority interrupts ticket with lower priority"] = tc
+	///////////////////////////////////////////////////////////////////
+
+	for name, tc := range testCases {
+		//uncomment this block to test individual test case
+		//if name != "test case 10: ticket with higher priority interrupts ticket with lower priority" {
+		//	continue
+		//}
+		fmt.Printf("testing %v\n", name)
+		s.runVolumeAttachmentTestCase(c, tc)
+	}
+
+}
+
+func (s *TestSuite) runVolumeAttachmentTestCase(c *C, tc *volumeAttachmentTestCase) {
+	kubeClient := fake.NewSimpleClientset()
+	lhClient := lhfake.NewSimpleClientset()
+	extensionsClient := apiextensionsfake.NewSimpleClientset()
+	kubeInformerFactory := informers.NewSharedInformerFactory(kubeClient, 0)
+	lhInformerFactory := lhinformerfactory.NewSharedInformerFactory(lhClient, 0)
+	ds := datastore.NewDataStore(lhInformerFactory, lhClient, kubeInformerFactory, kubeClient, extensionsClient, TestNamespace)
+	logger := logrus.StandardLogger()
+
+	volumeIndexer := lhInformerFactory.Longhorn().V1beta2().Volumes().Informer().GetIndexer()
+	volumeAttachmentIndexer := lhInformerFactory.Longhorn().V1beta2().VolumeAttachments().Informer().GetIndexer()
+
+	vac := NewLonghornVolumeAttachmentController(logger, ds, scheme.Scheme, kubeClient, TestOwnerID1, TestNamespace)
+	fakeRecorder := record.NewFakeRecorder(100)
+	vac.eventRecorder = fakeRecorder
+	for index := range vac.cacheSyncs {
+		vac.cacheSyncs[index] = alwaysReady
+	}
+
+	// Seed the data.
+	// Need to put it into both fakeclientset and Indexer because
+	// the fake client doesn't work well with informers.
+	// See details at https://github.com/kubernetes/kubernetes/issues/95372
+	vol, err := lhClient.LonghornV1beta2().Volumes(TestNamespace).Create(context.TODO(), tc.vol, metav1.CreateOptions{})
+	c.Assert(err, IsNil)
+	err = volumeIndexer.Add(vol)
+	c.Assert(err, IsNil)
+
+	volAttachment, err := lhClient.LonghornV1beta2().VolumeAttachments(TestNamespace).Create(context.TODO(), tc.volAttachment, metav1.CreateOptions{})
+	c.Assert(err, IsNil)
+	err = volumeAttachmentIndexer.Add(volAttachment)
+	c.Assert(err, IsNil)
+
+	////////////////////////////////////
+	// main test func
+	vac.syncHandler(getKey(volAttachment, c))
+	///////////////////////////////////
+
+	retVol, err := lhClient.LonghornV1beta2().Volumes(TestNamespace).Get(context.TODO(), tc.vol.Name, metav1.GetOptions{})
+	c.Assert(err, IsNil)
+	c.Assert(retVol.Spec, DeepEquals, tc.expectedVol.Spec)
+
+	retVolAttachment, err := lhClient.LonghornV1beta2().VolumeAttachments(TestNamespace).Get(context.TODO(), tc.volAttachment.Name, metav1.GetOptions{})
+	c.Assert(err, IsNil)
+	// mask timestamps
+	for _, ticketStatus := range retVolAttachment.Status.AttachmentTicketStatuses {
+		for ctype, condition := range ticketStatus.Conditions {
+			condition.LastTransitionTime = ""
```

Comment:
NIT: You can check `nowHandler` in other controllers and see how they handle this timestamp issue

Reply:
Thanks for the hint. I looked at this `nowHandler` before. I checked it again now. However, it doesn't seems to solve the timestamp of the conditions field. 

If we look into the [setCondition() function](https://github.com/longhorn/longhorn-manager/blob/a5b67e5ec57ec38be0d50cdbe77428d924f1b2fb/types/condition.go#L62), we can see that it directly `util.Now()` to set LastTransitionTime, it doesn't allow the caller to influence the behavior. This is why inside volume_controller_test.go, we also have to mask the LastTransitionTime of conditions [here](https://github.com/longhorn/longhorn-manager/blob/a5b67e5ec57ec38be0d50cdbe77428d924f1b2fb/controller/volume_controller_test.go#L1490) even though volume_controller has the `nowHandler`

I know perhaps, we can modify it so that setCondition accepts a `nowHandler` as an argument and use it to set time, but it would be outside scope of this unit test PR. In addition, I don't see much benefit from having a `nowHandler` function to set the condition LastTransitionTime to a fix a value comparing to just clearing them directly